tests/RobotFramework/tests/cumulocity/log/plugins/journald

```diff
@@ -0,0 +1,108 @@
+#!/bin/bash
+
+set -euo pipefail
+
+usage() {
+    echo "Usage: $0 <command> [args...]"
+    echo "Commands:"
+    echo "  list                                List all log types supported by this plugin"
+    echo "  get <type>                          Get logs for the specified type"
+    echo "      [--since <timestamp>]"
+    echo "      [--until <timestamp>]"
+    echo "      [--filter <text-filter>]"
+    echo "      [--tail <lines-from-end>]"
+    exit 1
+}
+
+list_log_types() {
+    echo ":::begin-tedge:::"
+    systemctl list-units --type=service --no-legend --no-pager --plain \
+        | awk '{ sub(/\.service$/, "", $1); print $1 }'
+    echo ":::end-tedge:::"
+}
+
+get_logs() {
+    local log_type="$1"
+    shift
+    
+    local since_arg=""
+    local until_arg=""
+    local filter_arg=""
+    local tail_arg=""
+    
+    while [[ $# -gt 0 ]]; do
+        case $1 in
+            --since)
+                since_arg="--since=@$2"
+                shift 2
+                ;;
+            --until)
+                until_arg="--until=@$2"
+                shift 2
+                ;;
+            --filter)
+                filter_arg="$2"
+                shift 2
+                ;;
+            --tail)
+                tail_arg="--lines=$2"
+                shift 2
+                ;;
+            *)
+                echo "Unknown option: $1" >&2
+                exit 1
+                ;;
+        esac
+    done
+    
+    local cmd="journalctl --no-pager --output=short-iso --unit=${log_type}"
+    
+    if [[ -n "$since_arg" ]]; then
+        cmd="$cmd $since_arg"
```

Comment:
I fully get that this version of the script if here mainly for testing purposes. However, the following raises some security concerns. The date argument being passed unchanged to the script, one can fool the plugin:

```
$ journald get tedge-agent --since '1758806229;ls /etc'
... #running the side command with sudo :-(
```

What can we do to prevent that? The script can protect itself and we might have to warn plugin devs:

```suggestion
        cmd="$cmd '$since_arg'"
```

Reply:
As noted here https://github.com/thin-edge/thin-edge.io/pull/3762#discussion_r2379633147, this is a false alarm. Yes a user might fool a plugin failing to protect itself against such attacks, but the same user cannot use the mapper to run as root a hacked log request.